osbs/build/build_request.py

```diff
@@ -748,6 +774,7 @@ def render(self, validate=True):
             del self.template['spec']['triggers']
 
         self.adjust_for_triggers()
+        self.adjust_for_scratch()
```

Comment:
If we ran `adjust_for_scratch()` before `adjust_for_triggers()`, we could actually remove any triggers for this build. We never want to have automated rebuilds for scratch builds.

(Note: this adjusts the BuildConfig, so kills any existing triggers... fixing that is separate issue, https://github.com/projectatomic/osbs-client/issues/370, which we won't worry about here)